test/app.test.js

```diff
@@ -77,4 +124,184 @@ describe('/account/sendRecoveryMessage', () => {
+
+        const response = await request.post('/account/seedPhraseAdded')
+            .send({ accountId, securityCode: 'lol', signature: 'wut' });
+
+        expect(response.status).toBe(403);
+        const account = await models.Account.findOne({ where: { accountId } });
+        expect(account).toBeFalsy();
+    });
+
+    test('finds/creates account, adds phraseAddedAt; returns recovery methods', async () => {
+        const accountId = await createNearAccount();
+        const { securityCode, signature } = await signatureFor(accountId);
+
+        const response = await request.post('/account/seedPhraseAdded')
+            .send({ accountId, securityCode, signature });
+
+        expect(response.status).toBe(200);
+        expect(response.body.phraseAddedAt).toBeTruthy();
+        const account = await models.Account.findOne({ where: { accountId } });
+        expect(account).toBeTruthy();
+    });
+});
+
+describe('/account/deleteRecoveryMethod', () => {
+    test('returns 400 (recoveryMethod invalid)', async () => {
+        const accountId = `account-${Date.now()}`;
+        await models.Account.create({ accountId });
+        const response = await request.post('/account/deleteRecoveryMethod')
+            .send({
+                accountId,
+                recoveryMethod: 'illegitimate',
+            });
+        expect(response.status).toBe(400);
+    });
+
+    test('returns 404 (accountId not found)', async () => {
+        const response = await request.post('/account/deleteRecoveryMethod')
+            .send({
+                accountId: 'illegitimate',
+                recoveryMethod: 'phone',
+            });
+        expect(response.status).toBe(404);
+    });
+
+    test('returns 403 Forbidden (signature not from accountId owner)', async () => {
+        const accountId = await createNearAccount();
+        await models.Account.create({ accountId, ...recoveryMethods });
+
+        let response = await request.post('/account/deleteRecoveryMethod')
+            .send({
+                accountId,
+                recoveryMethod: 'phone',
+                signature: 'lol',
+                securityCode: 'wut',
+            });
+
+        expect(response.status).toBe(403);
+    });
+
+    test('deletes specified recoveryMethod; returns recovery methods (account found, verified ownership, valid recoveryMethod)', async () => {
+        const accountId = await createNearAccount();
+        const account = await models.Account.create({ accountId, ...recoveryMethods });
+        const { securityCode, signature } = await signatureFor(accountId);
+
+        let response = await request.post('/account/deleteRecoveryMethod')
+            .send({
+                accountId,
+                recoveryMethod: 'phone',
+                signature,
+                securityCode,
+            });
+        expect(response.status).toBe(200);
+        await account.reload();
+        expect(response.body).toEqual(JSON.parse(JSON.stringify({
```

Comment:
why is `JSON.parse(JSON.stringify` roundtrip needed?

Reply:
Casts the date objects created by `new Date()` to strings – can you think of a more descriptive way to do this?

Reply:
I've done this in a new way now; I like it more and I think you will too!